### PR TITLE
Heroku builds queue functionality

### PR DESCRIPTION
**Pull request by ruslegg**

### CODE REVIEW

File: commands/shared/herokuBuildsExceeded.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+module.exports = async function herokuBuildsExceeded({ herokuKey, repo }) {
+  const currentTime = new Date().toISOString();
+  const headers = new Headers({
+    Accept: 'application/vnd.heroku+json; version=3',
+    Range: `started_at ${currentTime} ;order=desc,max=30;`,
+    Authorization: `Bearer ${herokuKey}`,
+  });
+  const res = await fetch(`https://api.heroku.com/apps/${repo}/builds`, {
+    headers,
+  });
```
Comment (ebdrup) [inline commands/shared/herokuBuildsExceeded.js:12]: add retries here. Networks are unstable.

File: program.js [hunk 2]
```diff
@@ -77,6 +77,8 @@ const program = programConstructor
         npmWhenFiles: program.npmWhenFiles,
         npmToken: process.env.NPM_TOKEN,
         slackToken: process.env.SLACK_TOKEN,
+        repo: process.env.CIRCLE_PROJECT_REPONAME,
+        herokuKey: process.env.HEROKU_API_KEY,
```
Comment (jonatanpedersen) [inline program.js:81]: I would call it `herokuApiKey` to make it consistent with the name of the env var and as it is for the `slackToken` and `npmToken`.

Comment (ruslegg) [reply program.js]: makes sense, done!

File: commands/master.js [hunk 3]
```diff
@@ -6,16 +6,22 @@ const {
...
+  herokuBuildsExceeded,
 } = require('./shared');
 const { exists } = require('./shared/fs');
 
-module.exports = async function master({ branch, npmToken, npmWhenFiles }) {
+module.exports = async function master({ branch, herokuKey, npmToken, npmWhenFiles, repo }) {
   // check env vars
   Object.entries(arguments[0]).forEach(([key, value]) => {
-    if (!value && !['branch', 'npmWhenFiles'].includes(key)) {
+    if (!value && !['branch', 'npmWhenFiles', 'herokuKey'].includes(key)) {
```
Comment (jonatanpedersen) [inline commands/master.js]: This is a very convoluted way of checking function arguments, and it also does not really check that the arguments are valid, just thay they are not undefined.

**Review by jonatanpedersen — APPROVED** [review]

**Pull request closed without merging**